### DISCUSSION TITLE
NUCLEO-H7S3L8 Mongoose

### DISCUSSION THREAD

**Discussion by CarloMalvicini** [cesanta/mongoose#2764] [Ask questions, get answers]
Hi,
I'm trying to implement the mongoose on NUCLEO-H7S3L8 without FreeDOS.
I already got it working with NUCLEO-H743ZI, just using the nucleo-h743zi-cube-baremetal-builtin example.

Since the ethernet modules on the two micontrollers look very similar, I tried first to follow the tutorial "Simple Network Interface with STM32 Using Ethernet and Mongoose Network Library", setting MG_ENABLE_DRIVER_STM32H.
Based on a working LwIP_TCP_Echo_Client project, and after some trials, I used the same configurator ETH module settings, same MPU_Config() function, same section attributes to the mongoose descriptors (commenting out the descriptors generated by the configurator).
Additionally in mongoose.c the full descriptors are initialized to zero (because the .ld file specifies as NOLOAD the descriptors sections), the ETH software reset and MDC clock divider are excluded (already done by STM code).

However I don't have DHCP working, what I can observe is:
- The RMII_MDIO signal is visible with a scope, period about 420 ns is correct.  
- The function mg_tcpip_driver_stm32h_tx() (with some added printf) continuosly tries to send a discovery packet.
- The packet is not visible with WireShark (capture filter: udp portrange 67-68).
- In mg_tcpip_driver_stm32h_tx(), the 314 packet seen with printf seems reasonable.
- The RMII_TXD0 signal to the LAN8742A is visible with the scope, correct packet duration, but it seems that the content is not correct (two much zeroes as compared to printed bytes and similar scope capture from working LwIP).
  
So the questions are:
- What may I have missed?
- Has someone got a working example for mongoose on NUCLEO-STM32H7S3L8 without FreeDOS?

Thank you,
C.M.

**Comment by scaprile** [comment]
We don't support that microprocessor/microcontroller, so we have no idea on what it requires or not, Also, I don't know what do you mean by FreeDOS.
I don't know where you got that tutorial from, nor if it is correct, our tutorials are on our webpages
Please see our [documentation](https://mongoose.ws/documentation/),  and follow the guidelines in our [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples).
"Pretty similar" does not mean _the same_, is it a Synopsys IP? What version ? Did ST also modify it ? Again, we don't support that processor, Is it even a Cortex-M ? It may or may not work.
You are mixing STM code and our code, your mileage may vary. 
If you are a paying customer or a prospect, please contact Support as per your contract.
Thank you.

**Comment by CarloMalvicini** [comment]
Sorry, I wrote FreeDOS instead of FreeRTOS.
